README.md

```diff
@@ -44,6 +44,299 @@ This is an experimental framework that aims to combine the lessons learned from
+-   **Data fields** like strings to represent one or multiple texts or arrays for image data. These elements can be
+    arbitrary python objects.
+-   **Annotation fields** like labeled spans for entities or labeled tuples of spans for relations. These elements have
+    to be of a certain container type `AnnotationList` that is dynamically typed with the actual annotation type, e.g.
+    `entities: AnnotationList[LabeledSpan]`. Furthermore, annotation elements define one or multiple annotation `targets`.
+    An annotation target is either a data element or another annotation container. Internally, targets are used to construct the
+    annotation graph, i.e. data elements and annotation containers are the nodes and targets define the edges. The
+    annotation graph defines the (de-)serialization order and what is accessible from within an annotation. To
+    facilitate the setup of annotation containers, there is the `annotation_field()` method.
+-   **Other fields** to save metadata, ids, etc. They are not constrained in any way, but can not be accessed from within
+    annotations.
+
+<details>
+
+<summary>
+
+#### Example Document Model
+
+</summary>
+
+```python
+from typing import Optional
+from pytorch_ie.core import Document, AnnotationList, annotation_field
+from pytorch_ie.annotations import LabeledSpan, BinaryRelation, Label
+
+class MyDocument(Document):
+    # data fields (any field that is targeted by an annotation fields)
+    text: str
+    # annotation fields
+    entities: AnnotationList[LabeledSpan] = annotation_field(target="text")
+    relations: AnnotationList[BinaryRelation] = annotation_field(target="entities")
+    label: AnnotationList[Label] = annotation_field()
+    # other fields
+    doc_id: Optional[str] = None
+```
+
+Note that the `label` is a special annotation field that does not define a target because it belongs to the whole document.
+You can also have more complex constructs, like annotation fields that target multiple other fields by using
+`annotation_field(targets)` or `annotation_field(named_targets)`. The latter is useful if you want to access the
+targets by name from within the annotation, see below for an example.
+
+</details>
+
+#### Annotations
+
+There are several predefined **annotation types** in `pytorch_ie.annotations`, however, feel free to define your own.
+Annotations have to be dataclasses that subclass `pytorch_ie.core.Annotation`. They also need to be hashable and
+immutable. The following is a simple example:
+
+```python
+@dataclass(eq=True, frozen=True)
+class SimpleLabeledSpan(Annotation):
+    start: int
+    end: int
+    label: str
+```
+
+<details>
+<summary>
+
+##### Accessing Target Content
+
+</summary>
+
+We can expand the above example a little to have a nice string representation:
+
+```python
+@dataclass(eq=True, frozen=True)
+class LabeledSpan(Annotation):
+    start: int
+    end: int
+    label: str
+
+    def __str__(self) -> str:
+        if self.targets is None:
```

Comment:
here vs L140 - self.targets or self.target?

Reply:
both work in this case, `targets` is a bit more general because `target` raises an error if multiple targets are defined. but you are right, this causes confusion, I will change it